### PR TITLE
Stabilize noop_waker

### PR DESCRIPTION
**Pull request by eholk**
Tracking Issue: #98286

This is a handy feature that's been used widely in tests and example async code and it'd be nice to make it available to users.

cc @rust-lang/wg-async

### CODE REVIEW

File: library/core/src/task/wake.rs [hunk 1]
```diff
@@ -866,7 +863,8 @@ impl LocalWaker {
     /// ```
     #[inline]
     #[must_use]
-    #[unstable(feature = "noop_waker", issue = "98286")]
+    #[stable(feature = "noop_waker", since = "CURRENT_RUSTC_VERSION")]
+    #[rustc_const_stable(feature = "noop_waker", since = "CURRENT_RUSTC_VERSION")]
```
Comment (zachs18) [inline library/core/src/task/wake.rs:867]: `LocalWaker` is itself still unstable, so I think this function should remain unstable, just under that feature gate?
```suggestion
    #[unstable(feature = "local_waker", issue = "118959")]
```

Comment (eholk) [reply library/core/src/task/wake.rs]: Definitely, thanks for the suggestion!

**Review by yoshuawuyts — APPROVED** [review]
Stabilizing this seems reasonable to me! We spent an extended period discussing this in @rust-lang/wg-async triage earlier this year, and I believe we were generally in favor of moving forward with this API. Thank you for putting this up!

**Comment by Amanieu** [comment]
@rfcbot fcp merge

**Review by dtolnay — APPROVED** [review]
r=me with 1 fewer rustc_const_stable attribute

**Comment by matthiaskrgr** [comment]
@bors r-
https://github.com/rust-lang/rust/pull/133783#issuecomment-2513534284

**Comment by eholk** [comment]
@bors r=dtolnay rollup=never

I somehow managed not to include the change to `await_ready.rs` that removed the `noop_waker` feature and only included the change to the `.coverage` file. This is fixed now.

**Comment by eholk** [comment]
@bors r=dtolnay

Rebased after conflicting upstream changes.

**Comment by eholk** [comment]
@bors r=dtolnay

Rebased and blessed the tests.

**Comment by Zalathar** [comment]
Looks like more Windows flakiness.

@bors retry

**Comment by traviscross** [comment]
@bors retry

This PR is having an unfortunate go of it.

File: library/core/src/task/wake.rs [hunk 2]
```diff
@@ -60,7 +60,8 @@ impl RawWaker {
         RawWaker { data, vtable }
     }
 
-    #[unstable(feature = "noop_waker", issue = "98286")]
+    #[stable(feature = "noop_waker", since = "CURRENT_RUSTC_VERSION")]
+    #[rustc_const_stable(feature = "noop_waker", since = "CURRENT_RUSTC_VERSION")]
     const NOOP: RawWaker = {
```
[the remark this answers is not in the document]
Comment (jhpratt) [reply library/core/src/task/wake.rs:65]: @eholk Why was this resolved? Unless the diff is somehow wrong, this wasn't changed.

[the remark this answers is not in the document]
Comment (dtolnay) [reply library/core/src/task/wake.rs:65]: It was resolved in https://github.com/rust-lang/rust/commit/2f45ea552f2aae10fdd7d7c4d9df91a11fadafec before I approved this PR in https://github.com/rust-lang/rust/pull/133089#issuecomment-2512776189.

That commit got lost by the force push above https://github.com/rust-lang/rust/pull/133089#issuecomment-2515389532.

@eholk please follow up.

**Comment by jieyouxu** [comment]
This *also* spuriously failed on msvc after it failed mingw

File: library/core/src/task/wake.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (eholk) [reply library/core/src/task/wake.rs:65]: Whoops, sorry about that. I have a PR up here to remove it: https://github.com/rust-lang/rust/pull/134100

**Pull request merged**